src/lib.rs

```diff
@@ -182,50 +182,74 @@ pub mod mock;
 //
 // can't use both sync and async secret service
 //
-#[cfg(all(feature = "sync-secret-service", feature = "async-secret-service"))]
-compile_error!("This crate cannot use the secret-service both synchronously and asynchronously");
+#[cfg(any(
+    all(feature = "sync-secret-service", feature = "async-secret-service"),
+    all(
+        feature = "sync-persistent-keyutils",
+        feature = "async-persistent-keyutils",
+    )
+))]
+compile_error!("This crate cannot use both the sync and async versions of any credential store");
 
 //
 // pick the *nix keystore
 //
-
-#[cfg(all(target_os = "linux", feature = "linux-native"))]
+#[cfg(all(
```

Comment:
Revert this once you make the feature change noted in Cargo.toml (having the new stores depend on linux-native.